src/memkind_memtier.c

```diff
@@ -256,7 +256,7 @@ static inline void decrement_alloc_size(unsigned kind_id, size_t size)
     unsigned bucket_id = t_hash_64();
     long long old_talloc =
         memkind_atomic_decrement(t_alloc_size[kind_id][bucket_id], size);
-    if (old_talloc - size < -FLUSH_THRESHOLD)
+    if ((old_talloc - (long long)size) < -FLUSH_THRESHOLD)
```

Comment:
Does it fix the issue?

Reply:
Yes, it does. Will add a test confirming that.